tests/image-tests.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+set -ex
+
+cleanup()
+{
+    docker rm -f $1
+}
+
+verify_output()
+{
+  if [ $1 ]; then
+    echo "FAILED: " $3
+    cleanup $2
+    exit 1
+  else
+    echo "PASSED: " $3
+    cleanup $2
+  fi
+}
+
+run_tests() {
+  docker build -t c-daemon .
+
+  name=version-check
+  docker run -d --name $name c-daemon
+  verify_output "-n $(docker logs $name | grep ${1//\/})" $name "check the version"
+
+  name=debug-daemon
+  docker run -d --name $name c-daemon --loglevel debug
+  verify_output "-n $(docker logs $name | grep Debug)" $name "set the loglevel to debug"
+
+  name=debug-daemon-with-cmd
+  docker run -d --name $name c-daemon --loglevel debug
+  verify_output "-n $(docker logs $name | grep Debug)" $name "set the loglevel to debug using cmd override"
+
+  name=no-logs
+  docker run -d --name $name c-daemon --logfile /var/log/newrelic/newrelic.log
+  verify_output "-n $(docker exec $name sh -c 'exec cat /var/log/newrelic/newrelic.log')" $name "set the logfile to /var/log/newrelic/newrelic.log"
+
+  name=no-logs-with-cmd
+  docker run -d --name $name c-daemon /usr/bin/newrelic-daemon --logfile /var/log/newrelic/newrelic.log
+  verify_output "-n $(docker exec $name sh -c 'exec cat /var/log/newrelic/newrelic.log')" $name "set the logfile to /var/log/newrelic/newrelic.log using cmd override"
+
+  name=cfg-debug
+  echo "loglevel=debug" > newrelic.cfg
+  docker run -d --name $name -v $PWD/newrelic.cfg:/etc/newrelic/newrelic.cfg c-daemon --c /etc/newrelic/newrelic.cfg
+  verify_output "-n $(docker logs $name | grep Debug)" $name "set the loglevel to debug using cfg file"
+
+}
+
+versions=(*/)
+
+# directories that don't have a docker image in them
+nonDockerDirs=(tests/ docs/)
+
+# remove directories we don't want to test
+for target in "${nonDockerDirs[@]}"; do
+  for i in "${!versions[@]}"; do
+    if [[ ${versions[i]} = "$target" ]]; then
+      unset 'versions[i]'
+    fi
+  done
+done
+echo "${versions[@]}"
+
+for dir in "${versions[@]}"; do
+  cd $dir
```

Comment:
This line fails if multiple versions are present.

If in the root dir, we have a `9.2.0` and another, for example `9.3.1`, we correctly `cd` into `9.2.0`, however during the next iteration of the loop, we attempt to `cd` into `9.3.1` relative to `9.2.0`.

`pushd` and `popd` are a pair of tools that make directory chances in loops pretty slick.  For example, this handles your intended behavior well:
```
for dir in "${versions[@]}"; do
  pushd $dir
  run_tests $dir
  popd
done
```

`pushd` is essentially "`cd` into the given arg, but push my `$PWD` onto a directories stack"